src/server_state.cpp

```diff
@@ -705,17 +705,9 @@ void wsrep::server_state::sst_sent(const wsrep::gtid& gtid, int error)
     }
 }
 
-<<<<<<< HEAD
-void wsrep::server_state::sst_received(wsrep::client_service& cs,
-                                       int const error, bool* awaiting_callback)
-||||||| 344544d
-void wsrep::server_state::sst_received(wsrep::client_service& cs,
-                                       int const error)
-=======
 int wsrep::server_state::sst_received(wsrep::client_service& cs,
```

Comment:
Will  you also change the PXC side to do something with this return value, or that's a todo for later?

Reply:
The callstack is:
sst_joiner_thread() -> void wsrep_sst_complete() -> wsrep::server_state::sst_received()
We are neither checking the error code nor catching exception in any caller of sst_received(). So the process just exits if the exception is thrown.

After the change we can:
1. Leave it as is (I don't remember having this exception a single time)
2. Handle the error in sst_joiner_thread() and do unireg_abort() (as it is done in case of other troubles)

I would go with the 2nd option (implementation on the server side, not this PR)

Reply:
I agree with Kamil's idea of going with 2nd option. Can you please create a JIRA for this so that we don't lose track of this discussion.

Reply:
I also agree, that was my question - if we'll change that now, or if it's a later TODO (annd jira ticket)

Reply:
I will try to adjust the server code now. If there are more troubles with doing this, I will create a separate ticket.